lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -14,12 +14,22 @@ def hosts(wspace = workspace.name, non_dead = false, addresses = nil, search_ter
     end
   end
 
-  # TODO: Shouldn't this proxy to RemoteHostDataService#find_or_create_host ?
-  # It's currently skipping the "find" part
   def find_or_create_host(opts)
+    host = get_host(opts)
+    return host unless host.nil?
+
     report_host(opts)
   end
 
+  def get_host(opts)
+    begin
+      data_service = self.get_data_service()
+      data_service.get_host(opts)
+    rescue Exception => e
```

Comment:
I overlooked this earlier, but these were all changed to `rescue => e` recently.